lib/Sema/TypeCheckDeclOverride.cpp

```diff
@@ -1519,6 +1519,18 @@ static bool checkSingleOverride(ValueDecl *override, ValueDecl *base) {
     diagnoseOverrideForAvailability(override, base);
   }
 
+  // Overrides of NSObject.hashValue are deprecated; one should override
+  // NSObject.hash instead.
+  if (auto baseVar = dyn_cast<VarDecl>(base)) {
+    if (auto classDecl =
+          baseVar->getDeclContext()->getAsClassOrClassExtensionContext()) {
+      if (classDecl->getBaseName().userFacingName() == "NSObject" &&
+          baseVar->getBaseName().userFacingName() == "hashValue") {
```

Comment:
Nitpick: `classDecl->getName().is("NSObject") && baseVar->getName() == ctx.Id_hashValue`.